bin/light-base/src/runtime_service/sub_utils.rs

```diff
@@ -0,0 +1,408 @@
+// Smoldot
+// Copyright (C) 2019-2022  Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//! This module contains additional useful public methods on [`RuntimeService`] that aren't part
+//! of the "core" functionnalities of the service.
+
+use super::*;
+
+impl<TPlat: Platform> RuntimeService<TPlat> {
+    /// Returns the current runtime version, plus an unlimited stream that produces one item every
+    /// time the specs of the runtime of the best block are changed.
+    ///
+    /// The future returned by this function waits until the runtime is available. This can take
+    /// a long time.
+    ///
+    /// The stream can generate an `Err` if the runtime in the best block is invalid.
+    ///
+    /// The stream is infinite. In other words it is guaranteed to never return `None`.
+    pub async fn subscribe_runtime_version(
+        &self,
+    ) -> (
+        Result<executor::CoreVersion, RuntimeError>,
+        stream::BoxStream<'static, Result<executor::CoreVersion, RuntimeError>>,
+    ) {
+        let mut master_stream = stream::unfold(self.guarded.clone(), |guarded| async move {
+            let subscribe_all = Self::subscribe_all_inner(&guarded, 16, 32).await;
```

Comment:
magic numbers?

Reply:
Yeah, I agree it's not great. These functions are only used in order to respond to some JSON-RPC methods, and so we have no choice but to use magic numbers anyway. The question is just where to put these magic numbers.

I think these functions should be moved to the `json_rpc_service`. I'll do that in a follow-up PR if you don't mind.